Nightly build failure with cuda/10.0+clang/8 on Volta with -Werror: error: unused function 'cuda_launch_blocking'

The nightly cuda/10.0+clang/8 build began failing with -Werror due to an apparent unused function following merge of #4403

**Error output**:
```
20:27:11 /home/jenkins/jenkins-new/workspace/Kokkos_develop_kokkosdev2_clang8_cuda10_0_uvm/kokkos/core/src/Cuda/Kokkos_Cuda_Instance.cpp:133:6: error: unused function 'cuda_launch_blocking' [-Werror,-Wunused-function]
20:27:11 bool cuda_launch_blocking() {
20:27:11      ^
20:27:11 1 error generated when compiling for sm_70.
``` 

**Reproducer** (kokkos-dev-2):
```
module load sems-env sems-cmake/3.17.1 clang/8.0 cuda/10.0

$KOKKOS_PATH/generate_makefile.bash --with-devices=Cuda,Serial --arch=SNB,Volta70 --compiler=clang++ --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="14" --kokkos-path=$KOKKOS_PATH --with-options=disable_profiling,disable_deprecated_code --with-cuda-options=enable_lambda,force_uvm  --no-examples
```

@e10harvey do you have time to look?
@dalg24 if this is a cuda/10.0 quirk resulting in tripping the unused-function should this be addressed by testing, maybe a case for pragma instead of flagma :)

--- a/core/src/Cuda/Kokkos_Cuda_Instance.cpp
+++ b/core/src/Cuda/Kokkos_Cuda_Instance.cpp
@@ -129,16 +129,6 @@ int cuda_kernel_arch() {
   return arch;
 }
 
-#ifdef KOKKOS_ENABLE_CUDA_UVM
-bool cuda_launch_blocking() {
-  const char *env = getenv("CUDA_LAUNCH_BLOCKING");
-
-  if (env == nullptr) return false;
-
-  return std::stoi(env);
-}
-#endif
-
 }  // namespace
 
 void cuda_device_synchronize(const std::string &name) {

